src/cantools/database/can/formats/arxml/utils.py

```diff
@@ -15,31 +15,51 @@ def parse_number_string(in_string : str, allow_float : bool=False) \
     - Some ARXML editors seem to sometimes include a dot in integer
       numbers (e.g., they produce "123.0" instead of "123")
     """
+    ret: Union[None, int, float] = None
+    in_string = in_string.strip().lower()
+
+    if len(in_string) > 0:
+        # the string literals "true" and "false" are interpreted as 1 and 0
+        if in_string == 'true':
+            ret = 1
+
+        if in_string == 'false':
+            ret = 0
+
+        # parse hex strings
+        if in_string.startswith('0x'):
+            ret = float.fromhex(in_string)
+
+        # allow octal notation without an "o" after the leading 0
```

Comment:
I guess that comment is not correct. `int(x, 0)` does this, but it would complain about `"011"`. (Note: I like the python notation much better than the C one, but IIRC AUTOSAR goes with the latter...)

Reply:
@FlorianMickler: do you indent to fix the comment or do you want the PR merged as-is?

Reply:
I just noticed: This should be an `elif` instead of `if` because strings starting with `0x` also fulfill that condition, so in effect all strings prefixed with a `0` would be interpreted as octal...

Reply:
Hi, sorry for replying late again. I don't get your complaint about the comment. Maybe I am being stupid, but octal notation without an o (letter o) after the leading 0 (zero) is what this check is for.  Or  not?

Reply:
> @FlorianMickler: do you indent to fix the comment or do you want the PR merged as-is?

Well, I'd like to understand the problem first before making that decision ..

Reply:
ok, I had a sign error here: I missed the "not" ;)